### PR TITLE
Add site admin packages page

### PR DESCRIPTION
**Pull request by umpox**
Closes https://github.com/sourcegraph/sourcegraph/issues/47093

- Adds site admin "Packages" page.
  - Currently just lists all packages in a similar way to how we list repositories.
  - Only shows if one of the package flags are enabled

image

## Test plan

Tested locally

## App preview:

- [Web](https://sg-web-tr-site-admin-list-packages.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.01% (+0.31 kb) | 0.05% (+7.38 kb) | 0.06% (+7.07 kb) | 0.14% (+1) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/tr/site-admin-list-packages/compare-report.html;tab=live_object) for a full comparison between the commits 4a8e0a6 and 84e9b0b or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/site-admin/SiteAdminPackagesPage.tsx [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+const PackageNode: React.FunctionComponent<React.PropsWithChildren<PackageNodeProps>> = ({ node }) => {
+    const packageRepository = node.repository
+
+    return (
+        <li className="list-group-item px-0 py-2">
+            <div className="d-flex align-items-center justify-content-between">
+                <div>
+                    {packageRepository ? (
+                        <>
+                            <ExternalRepositoryIcon externalRepo={packageRepository.externalRepository} />
```
Comment (Strum355) [inline client/web/src/site-admin/SiteAdminPackagesPage.tsx:100]: We should be able to derive the external service from the scheme, and display the icon that way. Perhaps some placeholder text where the gitserver metadata is? (not sure about that one yet, wondering whether the difference in height should be avoided from a design pov or not)

In this case, the package repo reference has been added to the references table, but it hasnt been synced yet. Theres likely some additional nuance thatll be introduced yet, but Im still scoping that out again

image

Comment (umpox) [reply client/web/src/site-admin/SiteAdminPackagesPage.tsx]: Any thoughts on what text to display when the package hasn't yet been synced? I assume we'll still support blocking as that just updates the service config file.

I've made it easier on the eye with a min height, but would be valuable to explain to the user why the package doesn't have a clickable link

image

**Comment by umpox** [comment]
@Strum355 This is ready for review. Happy to merge this and iterate when the blocklist support is available

File: client/web/src/site-admin/SiteAdminPackagesPage.tsx (hunk 1 above)
Comment (Strum355) [reply client/web/src/site-admin/SiteAdminPackagesPage.tsx]: > I assume we'll still support blocking as that just updates the service config file.

Yup thats correct

> would be valuable to explain to the user why the package doesn't have a clickable link

Thats what I was thinking as well. Perhaps a short-form text along the lines of "Not yet synced" with an :information_source: or dotted underlying for a hover popover with a longer explanation? Too long might look weird repeated multiple times hence the hover popover.

If you want, you can always merge this as-is now and go over this with design or play around with it afterwards to see what works.  High quality through iteration and all that :smile:  Theres likely more we could surface here such as blocked status and whatever else might be relevant as they come up over the next while.

**Review by Strum355 — APPROVED** [review]

File: client/web/src/site-admin/SiteAdminPackagesPage.tsx (hunk 1 above)
Comment (umpox) [reply client/web/src/site-admin/SiteAdminPackagesPage.tsx]: 👍  Will merge and we can iterate again when we add the blocked behavior

**Pull request merged**